tests/cli-rustup.rs

```diff
@@ -1402,6 +1402,174 @@ fn file_override_with_archive() {
     });
 }
 
+#[test]
+fn file_override_toml_format() {
+    setup(&|config| {
+        expect_ok(config, &["rustup", "default", "stable"]);
+        expect_ok(
+            config,
+            &[
+                "rustup",
+                "toolchain",
+                "install",
+                "nightly-2015-01-01",
+                "--no-self-update",
+            ],
+        );
+
+        expect_stdout_ok(config, &["rustc", "--version"], "hash-stable-1.1.0");
+
+        let cwd = config.current_dir();
+        let toolchain_file = cwd.join("rust-toolchain");
+        raw::write_file(
+            &toolchain_file,
+            r#"
+[toolchain]
+channel = "nightly-2015-01-01"
+"#,
+        )
+        .unwrap();
+
+        expect_stdout_ok(config, &["rustc", "--version"], "hash-nightly-1");
+    });
+}
+
+#[test]
+fn file_override_toml_format_add_components_toolchain_not_installed() {
+    setup(&|config| {
+        expect_ok(config, &["rustup", "default", "stable"]);
+
+        let cwd = config.current_dir();
+        let toolchain_file = cwd.join("rust-toolchain");
+        raw::write_file(
+            &toolchain_file,
+            r#"
+[toolchain]
+channel = "nightly-2015-01-01"
+components = [ "rust-src" ]
+"#,
+        )
+        .unwrap();
+
+        expect_stdout_ok(config, &["rustc", "--version"], "hash-nightly-1");
+        expect_stdout_ok(
```

Comment:
Ideally if `channel` were optional, this test would **a)** not change channel since it's purpose is to test that the component is added and **b)** ensure that the component you select is *NOT* installed before the toolchain file is written out.  (Note, this is kinda like the below test)

Reply:
I completely agree with `b)`, I've changed all the related tests to check that :+1:. 

With respect to `a)`, in this particular case I was trying to test that if the toolchain file specifies a toolchain that is not installed it would be installed before adding the component. Let me know if I misunderstood how rustup works with respect to this specific case. 

I've renamed and reordered a bit the tests to try to make my intent more explicit. Also I took advantage of the optional channel to reduce some test cases.

Reply:
It's fine that you want to verify the toolchain gets installed; in that context, simply don't install the other toolchain too.  Let's try and keep the test as fast as it can be.